src/main/java/com/code_intelligence/jazzer/runtime/CoverageMap.java

```diff
@@ -96,9 +99,68 @@ public static void enlargeIfNeeded(int nextId) {
     }
   }
 
+  private static ConcurrentHashMap<Integer, Integer> edgeMap = new ConcurrentHashMap<Integer, Integer>();
+  private static int nextEdgeId = 0;
+  static ReentrantLock rl = new ReentrantLock();
+  private static Integer initializeEdgeForAndroid(int id) throws InterruptedException {
+    // while loop to run until id gets lock
+    boolean gotLock = false;
+    int result = -1;
+
+    try {
+      while (!gotLock) {
+        gotLock = rl.tryLock();
+        if (gotLock) {
+          //check if it has been set since
+          Integer mappedId = edgeMap.get(id);
+          if (mappedId != null) {
+            return mappedId;
+          }
+
+          result = nextEdgeId++;
+          edgeMap.put(id, result);
+        } else {
+          Thread.sleep(1);
+        }
+      }
+    }
+    finally {
+      rl.unlock();
+    }
+
+    return new Integer(result);
+  }
+
+  public static void recordCoverageRemapped(final int id){
+    Integer mappedId = edgeMap.get(id);
```

Comment:
The agent contains [this abstraction](https://github.com/CodeIntelligenceTesting/jazzer/blob/6b9a31a650441d068bbc92e4f94e7d2b85d79bcf/src/main/java/com/code_intelligence/jazzer/agent/RuntimeInstrumentor.kt#L206-L215) for a similar purpose (multi-threaded fuzzing). 

We are also thinking of supporting offline instrumentation with Jazzer on all platforms, so maybe this could be integrated in a more general way. But I can think about that while we get this working for Android.